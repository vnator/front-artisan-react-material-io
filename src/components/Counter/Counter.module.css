.Counter {
  --border: 1px solid var(--white-full);
  --counter-size: 110px;

  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
  flex-grow: 1;
  background-color: var(--first-dark);
}

.autoCounter {
  border: var(--border);
  padding: var(--space-base);
  border-bottom-width: 0;
  box-sizing: border-box;
  margin: 0;
  width: var(--counter-size);
  background: transparent;
  color: var(--white-full);
  cursor: pointer;
  transition: var(--transition-base);
}

.autoCounter:hover {
  background: var(--white-full);
  color: var(--black-full);
}

.counter {
 margin-bottom: calc(var(--space-large)*2);
 display: flex;
 align-items: center;
 height: 32px;
 border: var(--border);
 box-sizing: border-box;
 width: var(--counter-size);
}

.counterEvent {
  border: 0;
  background: none;
  color: white;
  width: 32px;
  font-weight: bold;
  font-size: 24px;
  height: 100%;
  box-sizing: border-box;
  cursor: pointer;
  transition: var(--transition-base);
}

.counterEvent:hover {
  background: var(--white-full);
  color: var(--black-full);
}

.counterLabel {
  color: var(--white-full);
  border: var(--border);
  border-top: 0;
  border-bottom: 0;
  height: 100%;
  display: flex;
  align-items: center;
  padding: var(--space-base);
  box-sizing: border-box;
}

