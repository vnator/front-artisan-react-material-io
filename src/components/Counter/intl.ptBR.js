const counter_ptBR = {
  title: 'Counter',
  button: 'one by one',
};

export { counter_ptBR };
