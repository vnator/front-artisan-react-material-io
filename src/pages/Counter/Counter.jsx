import React from 'react';
import { IconButton, Button, Typography, makeStyles } from '@material-ui/core';
import {
  RemoveCircleOutlineOutlined,
  AddCircleOutlineOutlined,
} from '@material-ui/icons';

import { useSelector, useDispatch } from 'react-redux';

import { Title } from '../../components/Title.css';

import { setCounter } from './redux/actions.js';
import { setCounterOneByOne } from './redux/thunks.js';
import { el } from './element.selectors';
import { useIntl } from 'react-intl';

const useStyles = makeStyles({
  counter: {
    margin: '50px 0',
    display: 'inline-flex',
    flexDirection: 'column',
  },
  box: {
    display: 'inline-flex',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
});

const Counter = () => {
  const { formatMessage } = useIntl();
  const classes = useStyles();
  const counter = useSelector(state => state.counter);
  const dispatch = useDispatch();

  return (
    <div className={classes.counter}>
      <Title variant="h4" color="primary">
        {formatMessage({ id: 'counter.title' })}
      </Title>
      <Button
        color="primary"
        variant="outlined"
        id={el.oneByOne}
        onClick={() => dispatch(setCounterOneByOne())}>
        {formatMessage({ id: 'counter.button' })}
      </Button>
      <div className={classes.box}>
        <IconButton
          color="primary"
          variant="outlined"
          id={el.minus}
          onClick={() => dispatch(setCounter(counter - 1))}>
          <RemoveCircleOutlineOutlined />
        </IconButton>
        <Typography variant="subtitle2" color="secondary">
          {counter}
        </Typography>
        <IconButton
          color="primary"
          variant="outlined"
          id={el.plus}
          onClick={() => dispatch(setCounter(counter + 1))}>
          <AddCircleOutlineOutlined />
        </IconButton>
      </div>
    </div>
  );
};

export { Counter };
