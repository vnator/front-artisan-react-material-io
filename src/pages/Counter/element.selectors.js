const el = Object.freeze({
  oneByOne: 'test-counter-one_by_one',
  plus: 'test-counter-plus',
  minus: 'test-counter-minus',
});

export { el };
