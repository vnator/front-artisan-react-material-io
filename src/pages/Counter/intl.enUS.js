const counter_enUS = {
  title: 'Counter',
  button: 'one by one',
  plus: 'add one more',
  minus: 'reduce one',
};

export { counter_enUS };
