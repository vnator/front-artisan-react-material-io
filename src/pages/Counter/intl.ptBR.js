const counter_ptBR = {
  title: 'Contador',
  button: 'um por um',
  plus: 'adicionar um',
  minus: 'reduzir em um',
};

export { counter_ptBR };
