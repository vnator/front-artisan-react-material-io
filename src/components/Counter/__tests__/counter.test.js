import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { spy } from 'sinon';

import { Counter } from '../Counter';
import { el } from '../element.selectors';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureMockStore([thunk]);

describe('Counter', () => {
  it('Snapshot', () => {
    const store = mockStore({
      couter: 0,
    });

    const counter = mount(
      <Provider store={store}>
        <Counter />
      </Provider>,
    );

    expect(counter).toMatchSnapshot();
  });
  it('onClick Add', () => {
    const initialState = 23;

    const store = mockStore({
      counter: initialState,
    });

    store.dispatch = spy();

    const counter = mount(
      <Provider store={store}>
        <Counter />
      </Provider>,
    );

    counter.find(`.${el.plus}`).simulate('click');

    expect(store.dispatch.args[0][0].payload).toBe(initialState + 1);
  });

  it('onClick minus', () => {
    const initialState = 21;

    const store = mockStore({
      counter: initialState,
    });

    store.dispatch = spy();

    const counter = mount(
      <Provider store={store}>
        <Counter />
      </Provider>,
    );

    counter.find(`.${el.minus}`).simulate('click');

    expect(store.dispatch.args[0][0].payload).toBe(initialState - 1);
  });
  it('onClick OneByOne', () => {
    const store = mockStore({
      counter: 0,
    });

    store.dispatch = spy();

    const counter = mount(
      <Provider store={store}>
        <Counter />
      </Provider>,
    );

    counter.find(`.${el.oneByOne}`).simulate('click');

    expect(store.dispatch.calledOnce).toBeTruthy();
  });
});
