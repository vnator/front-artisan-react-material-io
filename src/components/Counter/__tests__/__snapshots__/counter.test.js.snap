// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Snapshot 1`] = `ReactWrapper {}`;
